Revised the loading screen before we begin loading the game :)

Starystars67 opened the pull request:
<img width="1959" height="640" alt="image" src="https://github.com/user-attachments/assets/3c8f460f-e956-410b-aa8f-b2d9aaea3ab9" />

looking for feedback ofc :)

Starystars67 committed:
Revised the loading screen before we begin loading the game :)

TheHawk09 commented:
That looks good! Adding a border-radius of 2px to the progress bar would be beneficial. Additionally, I believe the "X" button would look better with a larger border-radius.

Starystars67 commented:
Would you be okay with sending an image of what you have in mind?

WiserTixx commented:
In my opinion the rounded progress bar looks better but the "X" should stay the same to match the rounding of the container.

roitto commented on ui/modModules/multiplayer/multiplayer.js:
```diff
@@ -693,16 +695,84 @@ function($scope, $state, $timeout, $mdDialog, $filter, ConfirmationDialog, toast
 		} else {
-			document.getElementById('LoadingStatus').innerText = data.message;
+			if (document.getElementById('LoadingStatus').innerText != data.message) console.log(data.message)
+			if (data.message == "done") {
+				document.getElementById('LoadingStatus').innerText = "Done";
+				lastModInfo = ''
+			} else {
+				document.getElementById('LoadingStatus').innerText = data.message;
+			}
```
If you use the `text-transform: capitalize;` style for the LoadingStatus element, you could do something like this instead:


```suggestion
			if (data.message == "done") {
				lastModInfo = '';
			}

			document.getElementById('LoadingStatus').innerText = data.message;
```

.. Or is there some case where we don't want to show the data.message with capitalized text?

Starystars67 replied:
I'm not sure on this one, what does everyone else think?

TheHawk09 replied:
It’s probably better to capitalize just the main words, that would fit BeamNG’s style more

WiserTixx replied:
There are some messages that are in lowercase. I suppose it could make debugging harder in some cases.

TheHawk09 replied:
True, that’s a good point. Maybe just capitalize the main words manually instead of forcing it with CSS. Still, most of them are capitalized anyway.

roitto commented on ui/modModules/multiplayer/multiplayer.js:
```diff
@@ -693,16 +695,84 @@ function($scope, $state, $timeout, $mdDialog, $filter, ConfirmationDialog, toast
+			// Update current downloading mod info and if complete then push this mod into the downloaded mods info
+			$scope.$apply(function() {
+				// Update or add the current mod being downloaded
+				let existingMod = vm.downloadingMods.find(mod => mod.name === modName);
```
No need to use `let`, since this variable is not reassigned.

```suggestion
				const existingMod = vm.downloadingMods.find(mod => mod.name === modName);
```

roitto commented on ui/modModules/multiplayer/multiplayer.js:
```diff
@@ -693,16 +695,84 @@ function($scope, $state, $timeout, $mdDialog, $filter, ConfirmationDialog, toast
+
+				// If we switched to a new mod, mark the last one as done
+				if (lastModInfo != '' && lastModInfo != modName) {
+					let lastMod = vm.downloadingMods.find(mod => mod.name === lastModInfo);
```
No need to use `let`, since this variable is not reassigned.

```suggestion
					const lastMod = vm.downloadingMods.find(mod => mod.name === lastModInfo);
```

roitto commented on ui/modModules/multiplayer/multiplayer.js:
```diff
@@ -693,16 +695,84 @@ function($scope, $state, $timeout, $mdDialog, $filter, ConfirmationDialog, toast
+			// Update current downloading mod info and if complete then push this mod into the downloaded mods info
+			$scope.$apply(function() {
+				// Update or add the current mod being downloaded
+				let existingMod = vm.downloadingMods.find(mod => mod.name === modName);
```
No need to use `let`, since this variable is not reassigned.

```suggestion
				const existingMod = vm.downloadingMods.find(mod => mod.name === modName);
```

roitto commented on ui/modModules/multiplayer/multiplayer.js:
```diff
@@ -693,16 +695,84 @@ function($scope, $state, $timeout, $mdDialog, $filter, ConfirmationDialog, toast
+					existingMod.speed = speed;
+				} else {
+					// add this new mod to the beginning of the array
+					vm.downloadingMods = [{ number: modNumber, name: modName, progress: progress, speed: speed }, ...vm.downloadingMods];
```
Dunno about angular handles reactivity, but would it be better to just do unshift so we don't need to create new array each time?


```suggestion
					vm.downloadingMods.unshift({ number: modNumber, name: modName, progress: progress, speed: speed });
```

Starystars67 replied:
Sadly it was reactivity that made me have to do this this way :(

roitto commented on ui/modModules/multiplayer/multiplayer.js:
```diff
@@ -693,16 +695,84 @@ function($scope, $state, $timeout, $mdDialog, $filter, ConfirmationDialog, toast
+					existingMod.progress = '100';
+					existingMod.speed = 'Downloaded';
+				} else {
+					vm.downloadingMods = [{ number: modNumber, name: modName, progress: '100', speed: 'Loading' }, ...vm.downloadingMods];
```
Same thing as above, maybe unshift here? Also Maybe the speed should be `Downloaded` here(?)


```suggestion
					vm.downloadingMods.unshift({ number: modNumber, name: modName, progress: '100', speed: 'Downloaded' });
```

Starystars67 replied:
I like the idea of Downloaded but there is a lacking in state for each mod from the launcher currently.
If we could address that I would love to make it Downloaded and then loaded / ready when it is loaded.

Will update it to Downloaded though in the meantime.

roitto commented on ui/modModules/multiplayer/multiplayer.js:
```diff
@@ -693,16 +695,84 @@ function($scope, $state, $timeout, $mdDialog, $filter, ConfirmationDialog, toast
+			});
 		} else {
-			document.getElementById('LoadingStatus').innerText = data.message;
+			if (document.getElementById('LoadingStatus').innerText != data.message) console.log(data.message)
```
Is it necessary to log here? If so, maybe do it like this for consistency?


```suggestion
			if (document.getElementById('LoadingStatus').innerText != data.message) {
				console.log(data.message);
			}
```

Starystars67 replied:
Ah good spot thank you!

roitto commented on ui/modModules/multiplayer/multiplayer.js:
```diff
@@ -693,16 +695,84 @@ function($scope, $state, $timeout, $mdDialog, $filter, ConfirmationDialog, toast
+				} else {
+					vm.downloadingMods = [{ number: modNumber, name: modName, progress: '100', speed: 'Loading' }, ...vm.downloadingMods];
+				}
+			});
 		} else {
```
Maybe do something like this here for some special micro optimization, so the DOM won't be queried twice? 😀 

```js
const loadingStatusElement = document.getElementById('LoadingStatus`);

// then do the loadingStatusElement.innerText and loadingStatusElement.removeAttribute stuff here..
```

roitto commented on ui/modModules/multiplayer/multiplayer.js:
```diff
@@ -693,16 +695,84 @@ function($scope, $state, $timeout, $mdDialog, $filter, ConfirmationDialog, toast
 	};
 
+	vm.downloadingMods = [];
+	var lastModInfo = ''
```
```suggestion
	var lastModInfo = '';
```

roitto commented on ui/modModules/multiplayer/multiplayer.js:
```diff
@@ -630,6 +630,8 @@ function($scope, $state, $timeout, $mdDialog, $filter, ConfirmationDialog, toast
 		document.getElementById('LoadingStatus').setAttribute("hidden", "hidden");
 		document.getElementById('LoadingServer').style.display = 'none';
+		//vm.downloadingMods = [];
+		lastModInfo = ''
```
```suggestion
		lastModInfo = '';
```

roitto commented on ui/modModules/multiplayer/multiplayer.js:
```diff
@@ -630,6 +630,8 @@ function($scope, $state, $timeout, $mdDialog, $filter, ConfirmationDialog, toast
 		document.getElementById('OriginalLoadingStatus').removeAttribute("hidden");
 		document.getElementById('LoadingStatus').setAttribute("hidden", "hidden");
 		document.getElementById('LoadingServer').style.display = 'none';
+		//vm.downloadingMods = [];
```
Why commented out? Should this array be cleared if the popup is closed?

roitto replied:
Also if you want more micro-optimizations `vm.downloadingMods.length = 0;` could work here so there won't be need to allocate a new array. (if angular can handle this :grin: )

roitto commented on ui/modModules/multiplayer/multiplayer.html:
```diff
@@ -3,12 +3,35 @@
+      </div>
+      <span ng-if="mod.progress == 100" style="color: green; font-size: 1em; margin-left: 10px; float: right;">
+        <md-icon class="material-icons" style="vertical-align: middle;">check_mark</md-icon>
+      </span>
```
Missing indentation

```suggestion
        <span style="flex: 1;">
          <code>{{mod.name}}</code>
          {{ ' - ' }}
          <span style="color: darkgray;font-size: 0.8em;">{{mod.speed}}</span>
        </span>
        <div style="flex: 2;" ng-if="mod.progress != 100">
          <md-progress-linear md-mode="determinate" value="{{mod.progress}}" class="md-accent"></md-progress-linear>
        </div>
        <span ng-if="mod.progress == 100" style="color: green; font-size: 1em; margin-left: 10px; float: right;">
          <md-icon class="material-icons" style="vertical-align: middle;">check_mark</md-icon>
        </span>
```

Starystars67 replied:
Strange, the indentation is fine on my side?

Starystars67 replied:
I see it now, my bad, thank you!

Starystars67 committed:
Remove unnecessary log.

Co-authored-by: P Roitto <roitto@users.noreply.github.com>

Starystars67 committed:
minor adjustments

Co-Authored-By: P Roitto <22368098+roitto@users.noreply.github.com>

Starystars67 committed:
Add mod number back.

Co-Authored-By: P Roitto <22368098+roitto@users.noreply.github.com>

Starystars67 committed:
Added reference to element rather than query each time.

Starystars67 committed:
Rounded styling added

Starystars67 committed:
Fix indenting!

Co-authored-by: P Roitto <roitto@users.noreply.github.com>

WiserTixx reviewed (changes requested):
This PR looks good but I have a few requests.

Could this grey box be hidden when there is nothing to display in it?
<img width="1182" height="181" alt="image" src="https://github.com/user-attachments/assets/b9ed2e91-edac-41b7-acfe-4e30b1bde119" />

Same goes for the divider when there is no extra message.

And I've noticed when you quit a server while your downloading and then join another server the old history is partially displayed. Though this may be the launcher's fault.

Starystars67 committed:
Style changes for Tixx :)

WiserTixx committed:
Improve stability and clarity of new download menu

Properly hide and clear elements and distinguish between "Downloaded" and "Loading"

WiserTixx committed:
Localize new download menu and update

WiserTixx committed:
Fix serverlist profile name color not clearing
